website/project/model.py

```diff
@@ -4000,16 +4003,45 @@ def approve(self, user):
         self.approval.approve(user)
         self.add_status_log(user, DraftRegistrationLog.APPROVED)
         self.approval.save()
+        self.checkin_files()
 
     def reject(self, user):
         self.approval.reject(user)
         self.add_status_log(user, DraftRegistrationLog.REJECTED)
         self.approval.save()
+        self.checkin_files()
 
     def add_status_log(self, user, action):
         log = DraftRegistrationLog(action=action, user=user, draft=self)
         log.save()
 
+    def get_metadata_files(self):
```

Comment:
I'm generally opposed to adding new things to `website/project/model.py` unless absolutely necessary. That being said, I'm not certain that there's a good way around this without a refactor to how registrations work. It *may* be possible to modify the `PreregCallbackMixin` in `website/project/sactions.py` instead.